tracer-core/src/main/java/org/zalando/tracer/DefaultTracer.java

```diff
@@ -45,7 +45,7 @@
     }
 
     @Override
-    public void start(final Function<String, String> provider) {
+    public Stoppable start(final Function<String, String> provider) {
```

Comment:
> From client POV code will be the same

No, the current version requires clients to know/see/acknowledge a new interface. I was thinking about it, why not even like this?

    public interface Tracer implements AutoCloseable {
        @Override
        default void close() {
            stop();
        }
    }